### PR TITLE
Dev/gnss post processing

### PR DESCRIPTION
**Pull request by adamkrawczyk**
Enabling GNSS post processing

### CODE REVIEW

File: Gems/ROS2/Code/Source/GNSS/ROS2GNSSSensorComponent.cpp [hunk 1]
```diff
@@ -124,10 +126,21 @@ namespace ROS2
             m_gnssMsg.altitude = currentPositionWGS84.m_altitude;
             m_gnssMsg.status.status = sensor_msgs::msg::NavSatStatus::STATUS_SBAS_FIX;
             m_gnssMsg.status.service = sensor_msgs::msg::NavSatStatus::SERVICE_GPS;
+
+            bool registeredPostProcessingSupport = false;
+            GNSSPostProcessingRequestBus::EventResult(registeredPostProcessingSupport, entityId, &GNSSPostProcessingRequests::IsReady);
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/GNSS/ROS2GNSSSensorComponent.cpp]: What is purpose of GNSSPostProcessingRequests::IsReady?
If there is no handler available bus event would have no effect.

File: Gems/ROS2/Code/Include/ROS2/GNSS/GNSSPostProcessingRequestBus.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        using BusIdType = AZ::EntityId;
+        static constexpr AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::ById;
+        static constexpr AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+
+        //! Apply post-processing function, if any implementations to the bus are in the entity.
+        //! @param gnss standard gnss message passed as a reference. It will be changed through post-processing.
+        virtual void ApplyPostProcessing(sensor_msgs::msg::NavSatFix& gnss) = 0;
+
+        //! Query whether registered postprocessing is ready.
+        virtual bool IsReady() = 0;
```
Comment (michalpelka) [inline Gems/ROS2/Code/Include/ROS2/GNSS/GNSSPostProcessingRequestBus.h]: I do not understand why it is needed. It is redundant to Handler activation/deactivation.

**Review by michalpelka — COMMENTED** [review]
I think it is enough for simpler use cases. Please provide me defense of the existence `IsReady()` method in GNSSPostProcessingRequests

**Comment by michalpelka** [comment]
It should be merged to development branch as well @pijaro

**Comment by pijaro** [comment]
@michalpelka agree.

@adamkrawczyk, after the review, please make another PR with this very branch to the main repo in the [`o3de` organization](https://github.com/o3de/o3de-extras/).

**Review by pijaro — APPROVED** [review]

**Pull request merged**